src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultImmutableConverter.cs

```diff
@@ -62,47 +60,65 @@ public string GetConstructingTypeName(string immutableCollectionTypeName)
                 case ImmutableSetGenericInterfaceTypeName:
                     return  ImmutableHashSetTypeName;
                 default:
-                    Debug.Fail("Unknown immutable: " + immutableCollectionTypeName);
-                    return null;
+                    // TODO: Refactor exception throw following serialization exception changes. 
+                    throw new NotSupportedException(SR.Format(SR.DeserializeTypeNotSupported, immutableCollectionTypeName));
```

Comment:
I assume this this collection type will always fail? I.e. there is not some default behavior that we should try first?

Reply:
Yes for now. It's `ImmutableArray` - no default behavior comes to mind. I plan to handle it specially in a follow up.